packages/navigator/src/browser/navigator-editor-sync.ts

```diff
@@ -0,0 +1,165 @@
+import { FileNavigatorModel } from "./navigator-model";
+import { NavigatorConfiguration, NavigatorPreferences } from "./navigator-preferences";
+
+@injectable()
+export class NavigatorEditorSynchronizer {
+
+    @inject(FileNavigatorModel)
+    private readonly fileNavigatorModel: FileNavigatorModel;
+
+    @inject(EditorManager)
+    private readonly editorManager: EditorManager;
+
+    @inject(NavigatorPreferences)
+    private readonly navigatorPreferences: NavigatorPreferences;
+
+    private activeEditorChangedSubscription: Disposable | undefined;
+
+    @postConstruct()
+    protected async init() {
+        await this.navigatorPreferences.ready;
+        this.linkWithEditorPreferenceChanged(this.navigatorPreferences['navigator.linkWithEditor']);
+        this.navigatorPreferences.onPreferenceChanged(preference => this.onPreferenceChangedHandler(preference));
+    }
+
+    private onPreferenceChangedHandler(preference: PreferenceChangeEvent<NavigatorConfiguration>) {
+        if (preference.preferenceName === 'navigator.linkWithEditor') {
+            this.linkWithEditorPreferenceChanged(preference.newValue);
+        }
+    }
+
+    private linkWithEditorPreferenceChanged(linkWithEditorNewValue: boolean | undefined) {
+        if (linkWithEditorNewValue) {
+            if (this.activeEditorChangedSubscription === undefined) {
+                this.activeEditorChangedSubscription = this.editorManager.onActiveEditorChanged(
+                    (editor: EditorWidget | undefined) => this.selectNodeByEditor(editor));
+            }
+        } else {
+            if (this.activeEditorChangedSubscription !== undefined) {
+                this.activeEditorChangedSubscription.dispose();
+                this.activeEditorChangedSubscription = undefined;
+            }
+        }
+    }
+
+    /**
+     * Reveals and selects corresponding to given editor node in the navigator.
+     * If editor is undefined nothing happens.
+     *
+     * @param editor editor widget object to reveal its node in the navigator
+     */
+    selectNodeByEditor(editor: EditorWidget | undefined) {
+        if (editor) {
+            this.selectNodeById(this.editorIdToNavigatorNodeId(editor.id));
+        }
+    }
+
+    /**
+     * Converts editor id to navigator node id.
+     * Example: 'code-editor-opener:file:///home/user/workspace/README.md' => 'file:///home/user/workspace/README.md'
+     *
+     * @param editorId id of editor tab
+     * @returns id of corresponding navigator node
+     */
+    private editorIdToNavigatorNodeId(editorId: string) {
+        return editorId.substring(editorId.indexOf(':') + 1);
+    }
+
+    /**
+     * Reveals and selects node in the navigator by node id.
+     * If node with given id doesn't exist, nothing happens.
+     * Node id example: 'file:///home/user/workspace/README.md'
+     *
+     * @param nodeId navigator tree node id
+     */
+    selectNodeById(nodeId: string): void {
```

Comment:
It would be great, but I don't see a general way for now.
It could be done for existing tree structure like in `revealNode`, but this approach won't work for navigator, because target node might be not mounted (unless mount all the nodes which is not case), i.e. `getNode(id)` will return undefined because node isn't added into model (Tree.nodes) yet.

Reply:
ok, the first step will be to implement in for the navigator model. So it can be called by other clients.